qemu/tests/boot_with_different_vectors.py

```diff
@@ -0,0 +1,126 @@
+            mq_set_cmd = "ethtool -L %s combined %d" % (ifname, queues)
+            status, output = session.cmd_status_output(mq_set_cmd)
+            if status:
+                msg = "Fail to enable multi queue in guest."
+                msg += "Command %s, fail with output %s" % (mq_set_cmd, output)
+                error.TestError(msg)
+
+    def check_msi_support(session):
+        devices = session.cmd_output("lspci |grep Eth").strip()
+        vectors = params["vectors"]
+        error.context("Check if vnic inside guest support msi.", logging.info)
+        for device in devices.split("\n"):
+            if not device:
+                continue
+            d_id = device.split()[0]
+            msi_check_cmd = "lspci -vvv -s %s |grep MSI" % d_id
+            output = session.cmd_output(msi_check_cmd)
+            if vectors == 0 and output:
+                error.TestFail("Guest do not support msi.")
+            if output:
+                if vectors == 1:
+                    if "MSI-X: Enable-" in output:
+                        logging.info("MSI-X is disabled")
+                    else:
+                        msg = "Command %s get wrong output." % msi_check_cmd
+                        msg += " when vectors = 1"
+                        error.TestFail(msg)
+                else:
+                    if "MSI-X: Enable+" in output:
+                        logging.info("MSI-X is enabled")
+                    else:
+                        msg = "Command %s get wrong output." % msi_check_cmd
+                        msg += " when vectors = %s" % vectors
+                        error.TestFail(msg)
+
+    def check_interrupt(session, vectors):
+        error.context("Check the cpu interrupt of virito", logging.info)
+        cmd = "cat /proc/interrupts |grep virtio"
+        output = session.cmd_output(cmd)
+        vectors = int(vectors)
+        if vectors == 0 or vectors == 1:
+            if "IO-APIC-fasteoi" not in output:
+                msg = "Could not find IO-APIC-fasteoi interrupt"
+                msg += " when vectors = %s" % vectors
+                error.TestFail(msg)
+        elif 2 <= vectors and vectors <= 8:
+            if not re.findall("vritio[0-9]-virtqueues", output):
+                msg = "Could not find the device for msi interrupt "
+                msg += "when vectors = %s " % vectors
+                msg += "Command %s got output %s" % (cmd, output)
+                error.TestFail(msg)
+        elif vectors == 9 or vectors == 10:
+            if not (re.findall("virtio[0-9]-input", output) and
+                    re.findall("virtio[0-9]-output", output)):
+                msg = "Could not find the device for msi interrupt "
+                msg += "when vectors = %s " % vectors
+                msg += "Command %s got output %s" % (cmd, output)
+                error.TestFail(msg)
+
+    vectors_list = params["vectors_list"]
+    random.seed()
+    rand_num = random.randint(3, 7)
```

Comment:
Discuss with owner, auto script could test all the vectors from 0 to 11 and -1.
So update the script.